packages/neon/neon/lib/src/utils/global_options.dart

```diff
@@ -101,10 +101,6 @@ class GlobalOptions extends OptionsCollection {
         (final account) => MapEntry(account.id, (final context) => account.humanReadableID),
       ),
     );
-
-    if (!initialAccount.values.containsKey(initialAccount.value)) {
-      initialAccount.reset();
-    }
```

Comment:
This is problematic.
https://github.com/nextcloud/neon/blob/730d5791385478cc26d97e2cdcc2c1cd82626e4d/packages/neon/neon/lib/src/utils/global_options.dart#L30-L37

Sets the value to the first account if the option `rememberLastUsedAccount` is disabled.
If I have at least three accounts logged in and one of them set as the default one.
Then removing this default account leads to the value resetting causing `rememberLastUsedAccount` to be false and the value of `initialAccount` being null. 
I couldn't find any issue with this state but it's not the best to be in.

Reply:
Thinking about this more this issue should also be reproducible on main as we don't do anything different there.
If this is the case we should open a new issue.

Reply:
I tested it and it works fine, but as you said the state is not great. Instead of recovering from it the next time it should be recovered from immediately.